modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/StatisticsEndpoint.java

```diff
@@ -0,0 +1,378 @@
+      return to;
+    }
+
+    public String getResourceId() {
+      return resourceId;
+    }
+
+    public DataResolution getDataResolution() {
+      return dataResolution;
+    }
+  }
+
+  @GET
+  @Path("providers.json")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RestQuery(name = "getprovidersbyresourcetype", description = "Returns the available statistics providers for an (optional) resource type", returnDescription = "The available statistics providers as JSON", restParameters = {
+    @RestParameter(name = "resourceType", description = "The resource type: either 'episode', 'series' or 'organization'", isRequired = false, type = STRING)},
+    reponses = {
+      @RestResponse(description = "Returns the providers for the given resource type as JSON, or all, if the resource type is missing", responseCode = HttpServletResponse.SC_OK),
+      @RestResponse(description = "If the current user is not authorized to perform this action", responseCode = HttpServletResponse.SC_UNAUTHORIZED)
+    })
+  public Response getProviders(
+    @QueryParam("resourceType") final String resourceTypeStr) {
+    ResourceType resourceType;
+    try {
+      if (resourceTypeStr == null) {
+        resourceType = null;
+      } else {
+        resourceType = Enum.valueOf(ResourceType.class, resourceTypeStr.toUpperCase());
+      }
+    } catch (IllegalArgumentException e) {
+      return RestUtil.R.badRequest("invalid value for 'resourceType'");
+    }
+
+    JSONArray result = new JSONArray();
+    statisticsService
+      .getProviders(resourceType)
+      .stream()
+      .map(this::providerToJson)
+      .forEach(result::add);
+    return Response.ok(result.toJSONString()).build();
+  }
+
+  private static String providerTypeString(StatisticsProvider provider) {
+    if (provider instanceof TimeSeriesProvider) {
+      return TIME_SERIES_PROVIDER_TYPE;
+    }
+    return "unknown";
+  }
+
+  private JSONObject providerToJson(StatisticsProvider provider) {
+    final JSONObject providerObj = new JSONObject();
+    providerObj.put("providerId", provider.getId());
+    providerObj.put("providerType", providerTypeString(provider));
+    providerObj.put("title", provider.getTitle());
+    if (provider instanceof TimeSeriesProvider) {
+      providerObj.put("dataResolutions", resolutionsToJson(((TimeSeriesProvider) provider).getDataResolutions()));
+    }
+    providerObj.put("description", provider.getDescription());
+    return providerObj;
+  }
+
+  private JSONArray resolutionsToJson(Set<DataResolution> resolutions) {
+    JSONArray result = new JSONArray();
+    for (DataResolution dataResolution : resolutions) {
+      result.add(dataResolutionToJson(dataResolution));
+    }
+    return result;
+  }
+
+  private String dataResolutionToJson(DataResolution dataResolution) {
+    return dataResolution.toString().toLowerCase();
+  }
+
+  private static Optional<DataResolution> dataResolutionFromJson(final String dataResolution) {
+    try {
```

Comment:
Isn't this similar to what DataResolutionUtil does?

Reply:
I think here it is the same as above. We want to keep the admin-ui API and the external API separated.